src/opm/parser/eclipse/EclipseState/Aquifer/NumericalAquifer/NumericalAquifers.cpp

```diff
@@ -0,0 +1,141 @@
+    }
+
+    bool NumericalAquifers::hasCell(const size_t cell_global_index) const {
+        const auto& cells = this->aquifer_cells_;
+        return (cells.find(cell_global_index) != cells.end());
+    }
+
+    const NumericalAquiferCell& NumericalAquifers::getCell(const size_t cell_global_index) const {
+        assert(this->hasCell(cell_global_index));
+        return *(this->aquifer_cells_.at(cell_global_index));
+    }
+
+
+    bool NumericalAquifers::hasAquifer(const size_t aquifer_id) const {
+        return (this->aquifers_.find(aquifer_id) != this->aquifers_.end());
+    }
+
+    void NumericalAquifers::addAquiferConnections(const Deck& deck, const EclipseGrid& grid) {
+        const auto aquifer_connections = NumericalAquiferConnection::generateConnections(deck, grid);
+
+        for (auto& pair : this->aquifers_) {
+            const size_t aqu_id = pair.first;
+            const auto& aqu_cons = aquifer_connections.find(aqu_id);
+            if (aqu_cons == aquifer_connections.end()) {
+                const auto error = fmt::format("Numerical aquifer {} does not have any connections\n", aqu_id);
+                throw std::runtime_error(error);
+            }
+            auto& aquifer = pair.second;
+            const auto& cons = aqu_cons->second;
+
+            // For now, there is no two aquifers can be connected to one cell
+            // aquifer can not connect to aquifer cells
+            for (const auto& con : cons) {
+                const auto& aqu_con = con.second;
+                const size_t con_global_index = aqu_con.global_index;
+                if (this->hasCell(con_global_index)) {
+                    const size_t cell_aquifer_id = this->getCell(con_global_index).aquifer_id;
+                    auto msg = fmt::format("Problem with keyword AQUCON \n"
+                                           "Aquifer connection declared at grid cell ({}, {}, {}), is a aquifer cell "
+                                           "of Aquifer {}, and will be removed",
+                                           aqu_con.I + 1, aqu_con.J + 1, aqu_con.K + 1, cell_aquifer_id);
+                    OpmLog::warning(msg);
+                    continue;
+                }
+
+                aquifer.addAquiferConnection(con.second);
+            }
+        }
+    }
+
+    bool NumericalAquifers::operator==(const NumericalAquifers& other) const {
+        return this->aquifers_ == other.aquifers_;
+    }
+
+    size_t NumericalAquifers::numAquifer() const {
+        return this->aquifers_.size();
+    }
+
+    NumericalAquifers NumericalAquifers::serializeObject() {
+        NumericalAquifers result;
+        result.aquifers_  = {{1, SingleNumericalAquifer{1}}};
+        return result;
+    }
+
+    const SingleNumericalAquifer& NumericalAquifers::getAquifer(const size_t aquifer_id) const {
+        if ( !this->hasAquifer(aquifer_id)) {
```

Comment:
The "check-and-return-if-found" function is most efficiently implemented with iterators and the `find()` method.